mobile-tester.py

```diff
@@ -11,231 +11,265 @@
+  headers = {'Content-Type': 'application/octet-stream'}
+
+  logging.info("Uploading app %s", application_path)
+
+  app_data = open(application_path, 'rb').read()
+  requests.put(url=app_upload_url, data=app_data, headers=headers)
+  app_upload_succeeded = False
+  while not app_upload_succeeded:
+    requests.post(
+      url='{}/tests/{}/mobile-tests/check-uploads'.format(
+        base_url, test_id),
+      auth=auth,
+      params={
+        "fileType": "app"
+      })
+    test_object = requests.get(
+      url='{}/tests/{}'.format(base_url, test_id),
+      auth=auth).json()['result']
+    mobile_test_configuration = test_object['configuration']['plugins'][
+      'mobileTest']
+
+    if 'appUploadStatus' in mobile_test_configuration and mobile_test_configuration['appUploadStatus'] == 'SUCCEEDED':
+      app_upload_succeeded = True
+    else:
+      # validating application
+      logging.info('Waiting for verification')
+      time.sleep(SLEEP_TIME + 4)
 
-    response = requests.post(
+  # uploading tests...
+  if not silent:
+    test_package = raw_input(
+      "Enter zipped test files path, hit `enter` for default ({}): ".
+        format(test_files_path)).strip()
+    if len(test_package) == 0:
+      test_package = test_files_path
+  else:
+    test_package = test_files_path
+
+  logging.info("Uploading test package %s", test_package)
+  test_package_data = open(test_package, 'rb').read()
+  requests.put(url=test_package_url, data=test_package_data, headers=headers)
+  test_package_upload_succeeded = False
+  while not test_package_upload_succeeded:
+    requests.post(
+      url='{}/tests/{}/mobile-tests/check-uploads'.format(
+        base_url, test_id),
+      auth=auth,
+      params={
+        "fileType": "testPackage"
+      })
+    test_object = requests.get(
+      url='{}/tests/{}'.format(base_url, test_id),
+      auth=auth).json()['result']
+    mobile_test_configuration = test_object['configuration']['plugins'][
+      'mobileTest']
+    if 'testPackageUploadStatus' in mobile_test_configuration and mobile_test_configuration[
+      'testPackageUploadStatus'] == 'SUCCEEDED':
+      test_package_upload_succeeded = True
+    else:
+      # validating tests
+      logging.info('Waiting for verification')
+      time.sleep(SLEEP_TIME + 4)
+
+  # selecting device...
+  if not silent:
+    print("*** Available devices ****")
+    for i, device in enumerate(ios_devices[:10]):
+      print("{0:2d}. {1}({2})".format(i + 1, device['name'], device['os']))
+
+    choice = raw_input("Enter a choice for device, hit `enter` for default ({}({}): ".
+        format(chosen_device['name'], chosen_device['os'])).strip()
+    if len(choice) != 0:
+      device = ios_devices[int(choice) - 1]
+      logging.info("Chosen device is - %s(%s)",
+                   device['name'], device['os'])
+
+      data["configuration"]["plugins"]["mobileTest"][
+        "devices"][0]["arn"] = device['arn']
+
+      response = requests.post(
```

Comment:
@aryeguy  - I need your help as I don't think the put request with a newly selected device by user is working. Can we discuss this tomorrow morning your time?